examples/cuda/examples/maxwell/cpp.cuh

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright (c) NVIDIA
+ *
+ * Licensed under the Apache License Version 2.0 with LLVM Exceptions
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *   https://llvm.org/LICENSE.txt
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include "common.cuh"
+
+void run_cpp(float dt, bool write_vtk, std::size_t n_inner_iterations,
+             std::size_t n_outer_iterations, grid_t &grid,
+             std::string_view method) {
+  fields_accessor accessor = grid.accessor();
+
+  auto initializer = grid_initializer(dt, accessor);
+  for (std::size_t i = 0; i < accessor.cells; i++) {
+    initializer(i);
+  }
+
+  report_performance(grid.cells, n_inner_iterations * n_outer_iterations, method, 
+                     [&]() {
```

Comment:
Can we move the lambda out of the function call so that the indentation is less horrible